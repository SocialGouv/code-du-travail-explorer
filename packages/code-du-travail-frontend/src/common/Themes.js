import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Link from "next/link";
import { Layers } from "react-feather";
import {
  Grid,
  GridCell,
  Container,
  Section,
  SectionTitle,
  Tile
} from "@socialgouv/react-ui";

export default class Themes extends React.Component {
  static propTypes = {
    isRoot: PropTypes.bool,
    themes: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        slug: PropTypes.string.isRequired
      }).isRequired
    )
  };

  static defaultProps = {
    isRoot: false
  };

  render() {
    const { isRoot, themes } = this.props;
    if (!(themes.length > 0)) {
      return null;
    }
    return (
      <Section>
        <Container>
          <Link href="/themes" passHref>
            <SectionTitle desc="Retrouvez tous nos contenus autour de grands thèmes">
              Thèmes
            </SectionTitle>
          </Link>
          <Grid>
            {themes.map(({ slug, title }) => (
              <StyledGridCell isRoot={isRoot} key={slug + title}>
                <Link href="/themes/[slug]" as={`/themes/${slug}`} passHref>
                  <Tile icon={Layers}>{title}</Tile>
                </Link>
              </StyledGridCell>
            ))}
          </Grid>
        </Container>
      </Section>
    );
  }
}

const StyledGridCell = styled(GridCell)`
  height: ${({ isRoot }) => (isRoot ? "120px" : "auto")};
`;
