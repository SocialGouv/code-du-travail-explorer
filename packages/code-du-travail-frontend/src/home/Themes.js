import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { Link } from "../../routes";

import {
  Container,
  Grid,
  Category,
  GridCell,
  Section,
  theme,
  Wrapper
} from "@cdt/ui";

const iconsMap = {
  1: "hiring-1.svg",
  2: "remuneration.svg",
  3: "time.svg",
  4: "certificate.svg",
  5: "shield.svg",
  6: "handshake.svg",
  7: "file-3.svg",
  8: "book_web.svg"
};

export default class Themes extends React.Component {
  static propTypes = {
    themes: PropTypes.arrayOf(
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        slug: PropTypes.string.isRequired
      }).isRequired
    ),
    title: PropTypes.string
  };

  static defaultProps = {
    title: "Retrouvez nos réponses thématiques"
  };

  render() {
    const { title, themes } = this.props;
    if (!(themes.length > 0)) return null;
    return (
      <Section>
        <Container>
          <Wrapper>
            {title && <Title>{title}</Title>}
            <Grid>
              {themes.map(({ id, slug, label, parent }) => (
                <GridCell key={slug + label}>
                  <Link route="themes" params={{ slug: slug || "/" }} passHref>
                    <Tile title={label}>
                      <Category
                        small={parent}
                        title={label}
                        icon={`/static/assets/icons/${iconsMap[id] ||
                          "profiles.svg"}`}
                      />
                    </Tile>
                  </Link>
                </GridCell>
              ))}
            </Grid>
          </Wrapper>
        </Container>
      </Section>
    );
  }
}
const { box, spacing, colors } = theme;

const Title = styled.h2`
  text-align: center;
  margin-bottom: ${spacing.large};
`;

const Tile = styled.a`
  text-decoration: none;
  display: block;
  height: 100%;
  border-radius: ${box.borderRadius};
  & > * {
    transition: all 0.2s ease;
  }
  :focus > *,
  :active > *,
  :hover > * {
    transform: scale(1.1);
    border: 1px solid ${colors.focus};
    box-shadow: 0 0 2px 2px ${colors.focusShadow};
  }
`;
