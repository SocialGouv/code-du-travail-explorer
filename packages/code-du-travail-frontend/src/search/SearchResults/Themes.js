import React from "react";
import Link from "next/link";
import styled from "styled-components";
import { Layers } from "react-feather";

import { getRouteBySource } from "@cdt/sources";
import {
  CardList,
  Container,
  Section,
  Tile,
  Wrapper
} from "@socialgouv/react-ui";

export const Themes = ({ items, query }) => (
  <Section>
    <Container>
      <Wrapper>
        <CardList title="Les thèmes suivants peuvent vous intéresser">
          {items.map(({ slug, title, source }) => (
            <Link
              key={slug}
              href={{
                pathname: `${getRouteBySource(source)}/[slug]`,
                query: query ? { q: query } : null
              }}
              as={`/${getRouteBySource(source)}/${slug}${
                query ? `?q=${query}` : ""
              }`}
              passHref
            >
              <StyledTile icon={Layers}>{title}</StyledTile>
            </Link>
          ))}
        </CardList>
      </Wrapper>
    </Container>
  </Section>
);

const StyledTile = styled(Tile)`
  min-height: 120px;
`;
