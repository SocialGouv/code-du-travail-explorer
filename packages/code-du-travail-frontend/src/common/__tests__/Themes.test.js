import React from "react";
import { render } from "@wrapped-testing-library/react";
import Themes from "../Themes";
const themes = [
  {
    id: 1,
    title: "Recrutement et contrat de travail",
    slug: "1-recrutement-et-contrat-de-travail"
  },
  {
    id: 2,
    title: "Salaire et Rémunération",
    slug: "2-salaire-et-remuneration"
  },
  {
    id: 3,
    title: "Temps de travail et congés",
    slug: "3-temps-de-travail-et-conges"
  },
  {
    id: 4,
    title: "Emploi et formation",
    slug: "4-emploi-et-formation"
  },
  {
    id: 5,
    title: "Santé, sécurité et conditions de travail",
    slug: "5-sante,-securite-et-conditions-de-travail"
  },
  {
    id: 6,
    title: "Représentation du personnel et négociation collective",
    slug: "6-representation-du-personnel-et-negociation-collective"
  },
  {
    id: 7,
    title: "Départ de l'entreprise",
    slug: "7-depart-de-lentreprise"
  },
  {
    id: 8,
    title: "Conflits au travail et contrôle de la réglementation",
    slug: "8-conflits-au-travail-et-controle-de-la-reglementation"
  }
];

describe("<Themes />", () => {
  it("should render", () => {
    const { container } = render(<Themes themes={themes} />);
    expect(container).toMatchSnapshot();
  });
});
