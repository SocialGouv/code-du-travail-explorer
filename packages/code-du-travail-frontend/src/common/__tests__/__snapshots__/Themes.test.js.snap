// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Themes /> should render 1`] = `
.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c0 {
  padding: 1.25rem 0;
}

.c6 {
  padding: 0.625rem 1rem;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  text-align: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  color: #0053b3;
  font-size: 1.125rem;
}

.c7 {
  width: 2rem;
  display: inline-block;
  margin: 0.2rem 0.625rem 0.2rem 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  list-style-type: none;
  padding: 0;
  margin: NaN;
}

.c4 {
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-basis: calc(100% / 4 - 2 * 0.625rem);
  -ms-flex-preferred-size: calc(100% / 4 - 2 * 0.625rem);
  flex-basis: calc(100% / 4 - 2 * 0.625rem);
  margin: 0.625rem;
  text-align: center;
}

.c2 {
  text-align: center;
  margin-bottom: 2rem;
}

.c5 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: block;
  height: 100%;
  border-radius: 0.25rem;
}

.c5 > * {
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c5:focus > *,
.c5:active > *,
.c5:hover > * {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  border: 1px solid #80bdff;
  box-shadow: 0 0 2px 2px #bfdeff;
}

@media print {
  .c1 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:980px) {
  .c4 {
    -webkit-flex-basis: calc(100% / 3 - 2 * 0.625rem);
    -ms-flex-preferred-size: calc(100% / 3 - 2 * 0.625rem);
    flex-basis: calc(100% / 3 - 2 * 0.625rem);
  }
}

@media (max-width:600px) {
  .c4 {
    -webkit-flex-basis: calc(100% - 2 * 0.625rem);
    -ms-flex-preferred-size: calc(100% - 2 * 0.625rem);
    flex-basis: calc(100% - 2 * 0.625rem);
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        Retrouvez nos réponses thématiques
      </h2>
      <ul
        class="c3"
      >
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/1-recrutement-et-contrat-de-travail"
            title="Recrutement et contrat de travail"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/hiring-1.svg"
              />
              <h3
                class="c8"
              >
                Recrutement et contrat de travail
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/2-salaire-et-remuneration"
            title="Salaire et Rémunération"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/remuneration.svg"
              />
              <h3
                class="c8"
              >
                Salaire et Rémunération
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/3-temps-de-travail-et-conges"
            title="Temps de travail et congés"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/time.svg"
              />
              <h3
                class="c8"
              >
                Temps de travail et congés
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/4-emploi-et-formation"
            title="Emploi et formation"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/certificate.svg"
              />
              <h3
                class="c8"
              >
                Emploi et formation
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/5-sante,-securite-et-conditions-de-travail"
            title="Santé, sécurité et conditions de travail"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/shield.svg"
              />
              <h3
                class="c8"
              >
                Santé, sécurité et conditions de travail
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/6-representation-du-personnel-et-negociation-collective"
            title="Représentation du personnel et négociation collective"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/handshake.svg"
              />
              <h3
                class="c8"
              >
                Représentation du personnel et négociation collective
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/7-depart-de-lentreprise"
            title="Départ de l'entreprise"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/file-3.svg"
              />
              <h3
                class="c8"
              >
                Départ de l'entreprise
              </h3>
            </div>
          </a>
        </li>
        <li
          class="c4"
        >
          <a
            class="c5"
            href="/themes/8-conflits-au-travail-et-controle-de-la-reglementation"
            title="Conflits au travail et contrôle de la réglementation"
          >
            <div
              class="c6"
            >
              <img
                alt=""
                class="c7"
                src="/static/assets/icons/book_web.svg"
              />
              <h3
                class="c8"
              >
                Conflits au travail et contrôle de la réglementation
              </h3>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
